File: backend-microservices/notice-service/src/main/java/com/mrca/notice/model/Notice.java
package com.mrca.notice.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "notices")
@Data
public class Notice {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    @Column(nullable = false, length = 300)
    private String title;
    
    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "is_urgent")
    private Boolean isUrgent = false;
    
    @Column(name = "is_active")
    private Boolean isActive = true;
    
    @Column(name = "created_date")
    private LocalDateTime createdDate;
    
    @Column(name = "updated_date")
    private LocalDateTime updatedDate;
    
    @PrePersist
    protected void onCreate() {
        createdDate = LocalDateTime.now();
        updatedDate = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedDate = LocalDateTime.now();
    }
}

File: frontend-react/src/types/notice.ts
export interface Notice {
  id: number;
  title: string;
  description: string;
  is_urgent: boolean;
  is_active: boolean;
  created_date: string;
  updated_date: string;
}

export interface Update {
  id: number;
  message: string;
  is_active: boolean;
  created_date: string;
  updated_date: string;
}